### PR TITLE
lcms2

### PR DESCRIPTION
**Pull request by insertinterestingnamehere**
Needed as a dependency elsewhere.

### CODE REVIEW

File: recipes/lcms2/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cd $SRC_DIR
+
+./configure                     \
+            --prefix=$PREFIX    \
+            --with-jpeg=$PREFIX \
+            --with-tiff=$PREFIX \
+            --with-zlib=$PREFIX
+
+make
+make install
```
Comment (ocefpaf) [inline recipes/lcms2/build.sh]: Is there a `make check` we can run here?

File: recipes/lcms2/build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+cd $SRC_DIR
```
Comment (ocefpaf) [inline recipes/lcms2/build.sh]: I don't think you need this.

File: recipes/lcms2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  number: 0
+  features:
+    - vc9   # [win and py27]
+    - vc10  # [win and py34]
+    - vc14  # [win and py35]
+
+requirements:
+  build:
+    - cmake         # [win]
+    - jpeg 8d
```
Comment (ocefpaf) [inline recipes/lcms2/meta.yaml]: In `conda-forge` we are pinning to `jpeg 9*`. (See [here](https://github.com/conda-forge/staged-recipes/wiki/Pinned-dependencies).)

If you really need `jpeg 8d` it is fine but note that you won't be able to install this with the rest of the stack.

File: recipes/lcms2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - libtiff 4.0.6
```
Comment (ocefpaf) [inline recipes/lcms2/meta.yaml]: Unless you really need to be this specif `libtiff 4.0.*` is recommend. Same for below `zlib 1.2.*`

File: recipes/lcms2/build.sh (hunk 2 above)
Comment (msarahan) [inline recipes/lcms2/build.sh]: Yes, this is where conda-build drops you by default.

File: recipes/lcms2/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package:
+  name: lcms2
+  version: 2.7
+
+source:
+  fn: lcms2-2.7.tar.gz
```
Comment (patricksnape) [inline recipes/lcms2/meta.yaml]: To make updating the recipe simpler we usually use a jinja template for the version and then replace all instances of `2.7` with the template.

**Comment by insertinterestingnamehere** [comment]
Thanks for all the comments! I've addressed most of them in the last few commits.
Perhaps this is a matter of personal preference, but I like to totally fix the versions for dependencies in recipes I write. It helps make it clearer what library versions were used to build a given binary. The downside is that the build-time dependencies have to be manually updated, but I'm fine with doing that.

**Comment by ocefpaf** [comment]
> but I like to totally fix the versions for dependencies in recipes I write

That is fine as long as the package you are maintaining is not at the base of the stack. Something like that would not be acceptable in `numpy` for example, but as long as nothing depends on `lcms2` it is OK. The main reason is to avoid unnecessary downgrades and keep the stack consistent.

Note that, if you pinned them to the exact version at build time it would be nice to pin them the same way at run time to avoid ABI breakages. So instead of:

```yaml
    - jpeg >=9b
    - jpeg 9*
    - libtiff >=4.0.6
    - libtiff 4.0.*
    - zlib >=1.2.8
    - zlib 1.2.*
```

Can you do?

```yaml
    - jpeg 9b
    - libtiff 4.0.6
    - zlib 1.2.8
```

File: recipes/lcms2/meta.yaml [hunk 6]
```diff
@@ -17,12 +19,13 @@ build:
...
   build:
     - cmake         # [win]
-    - jpeg 8d
+    - jpeg 9b
     - libtiff 4.0.6
     - python        # [win]
     - zlib 1.2.8
   run:
-    - jpeg 8d
+    - jpeg >=9b
```
Comment (patricksnape) [inline recipes/lcms2/meta.yaml]: Jpeg is listed twice.

**Comment by insertinterestingnamehere** [comment]
Can you elaborate a bit more on how this would cause unnecessary downgrades? Completely fixing dependencies at build time doesn't effect the dependencies selected at runtime.

My reasoning with listing each package twice is that, in general, when a package is built against a given binary, the ABI backwards compatibility guarantees of each dependency make it so that binaries built with older versions of the package will still work as long as the binary is from the same release series. I included the two separate constraints for each dependency to accurately represent the fact that, at runtime the version used should be from the same release series (e.g. from the jpeg 9 series) and be at least as new as the version used at build time (e.g. later than jpeg 9b).

File: recipes/lcms2/bld.bat [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      -DJPEG_LIBRARY=%LIBRARY_LIB%\libjpeg.lib ^
+      -DJPEG_INCLUDE_DIR=%LIBRARY_INC%         ^
+      -DTIFF_LIBRARY=%LIBRARY_LIB%\libtiff.lib ^
+      -DTIFF_INCLUDE_DIR=%LIBRARY_INC%         ^
+      -DZLIB_LIBRARY=%LIBRARY_LIB%\zlib.lib    ^
+      -DZLIB_INCLUDE_DIR=%LIBRARY_INC%         ^
+      -DCMAKE_EXE_LINKER_FLAGS="/verbose:lib"  ^
+      ..
+if errorlevel 1 exit 1
+
```
Comment (jakirkham) [inline recipes/lcms2/bld.bat]: Can we run `ctest`?

File: recipes/lcms2/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,73 @@
+include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include")
```
Comment (jakirkham) [inline recipes/lcms2/CMakeLists.txt]: Is it possible for us to submit this upstream?

**Pull request closed without merging**